### PR TITLE
Build overhaul

### PR DESCRIPTION
**Pull request by weisJ**
See #412 

This time gradle is rewired to use the old source/resource locations.

The only thing which couldn't be avoided is to move the test resources into their own directory. This is due to the nature of the `java-library` package.

Github history is preserved this time: e.g. https://github.com/gurkenlabs/litiengine/commits/136a19900ca8d5a79b760d2c90ab6f9d0646be44/src/de/gurkenlabs/litiengine/Game.java

### CODE REVIEW

**Comment by weisJ** [comment]
This PR doesn’t include the automatic formatting through spotless. I can add it back in or create a separate PR.

**Comment by nightm4re94** [comment]
> This PR doesn’t include the automatic formatting through spotless. I can add it back in or create a separate PR.

I think it's best to open another PR for that once everything else is back in order. 
Thank you!

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Auto detect text files and perform LF normalization
+*        text=auto
+
+*.java   text
```
Comment (Gamebuster19901) [inline .gitattributes:4]: Currently, `.java` files use cl-rf in this repo. Keeping this as the standard would keep more of the `git blame history` intact.

Comment (weisJ) [reply .gitattributes]: Updated. Though there were still some files for which the line endings changed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -31,11 +31,11 @@ jobs:
             ~/.gradle/wrapper/'
           key: ${{ runner.os }}-mvn_sonar_gradle_cache
 
-      - name: Set up JDK 8
+      - name: Set up JDK 16
```
Comment (Gamebuster19901) [inline .github/workflows/build.yml:34]: Apologies I'm not fluent in Kolten; Is the name on line 20 still correct, or is the project getting built with java 16 now?

Comment (weisJ) [reply .github/workflows/build.yml:34]: Yes the project is now built using java 16.  In the discord I was told that this is something that was planned anyways. If desired I can build the engine itself with an older version. Though for `utiliti` it is easier if `jpackage` is available.

It is configured here:
https://github.com/weisJ/litiengine/blob/6c2023cc5d9883bd41ab175cfd48e88f069b12c4/build.gradle.kts#L60

Notably if no matching compiler and jdk is available gradle will download an appropriate version itself.

Comment (Gamebuster19901) [reply .github/workflows/build.yml:34]: Line #20 should also be updated to say `Java 16 build` instead of `Java 8 build` then.

Comment (weisJ) [reply .github/workflows/build.yml:34]: Fixed

**Comment by nightm4re94** [comment]
Thanks for addressing the open issues :) Is this PR now ready for review?

**Comment by weisJ** [comment]
Yes it is :)

**Comment by nightm4re94** [comment]
This has been a huge improvement - we really appreciate the effort you've put into this PR! <3

**Comment by weisJ** [comment]
I’d suggest providing a new version for utiliti with the jpackage build. It seems like there have been more and more issues with people not getting it to run with their Java version. The jpackage build bundles it’s own jre, so this shouldn’t be a problem anymore.

**Comment by nightm4re94** [comment]
Hi again @weisJ, we are currently trying to get a local clone of the engine imported as a Gradle dependency in a project.
Usually, we had used the following gradle configuration to make this work:

in `build.gradle`:
```groovy
dependencies {
    implementation project(':litiengine')
}
```

in `settings.gradle`:
```groovy
include ":litiengine"
project(":litiengine").projectDir = file("../litiengine")
```

As we have just found out, this setup is no longer working for our existing projects. What would the correct configuration be to use a local copy of the engine?

**Comment by weisJ** [comment]
Assuming the following project structure:
````
project/settings.gradle
litiengine/setting.gradle.kts
````
Then in `project/settings.gradle`:
````
includeBuild "../litiengine"
````
You'll be able to declare litiengine as an dependency as usual (i.e. as if you were consuming it from maven). Gradle will automatically replace any dependency with the version provided by `includeBuild` if it finds a matching module.
At first this will fail with the error message that two modules satisfy the requested dependency. Simply add the following line to 
`litiengine/setting.gradle.kts`:
````
rootProject.name = "litiengine-root"
````
It will disambiguate the module names.

**Comment by nightm4re94** [comment]
Thanks for clarifying!

**Pull request merged**